web/web.go

```diff
@@ -49,77 +50,66 @@ var (
 
 // WebService handles the HTTP endpoints with the exception of /api.
 type WebService struct {
+	QuitChan chan struct{}
+	router   *route.Router
+}
+
+type WebServiceOptions struct {
+	PathPrefix      string
 	StatusHandler   *PrometheusStatusHandler
 	MetricsHandler  *api.MetricsService
 	AlertsHandler   *AlertsHandler
 	ConsolesHandler *ConsolesHandler
 	GraphsHandler   *GraphsHandler
-
-	QuitChan chan struct{}
 }
 
-// ServeForever serves the HTTP endpoints and only returns upon errors.
-func (ws WebService) ServeForever(pathPrefix string) {
-
-	http.Handle("/favicon.ico", http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
-		http.Error(w, "", 404)
-	}))
-
-	http.HandleFunc("/", prometheus.InstrumentHandlerFunc(pathPrefix, func(rw http.ResponseWriter, req *http.Request) {
-		// The "/" pattern matches everything, so we need to check
-		// that we're at the root here.
-		if req.URL.Path == pathPrefix+"/" {
-			ws.StatusHandler.ServeHTTP(rw, req)
-		} else if req.URL.Path == pathPrefix {
-			http.Redirect(rw, req, pathPrefix+"/", http.StatusFound)
-		} else if !strings.HasPrefix(req.URL.Path, pathPrefix+"/") {
-			// We're running under a prefix but the user requested something
-			// outside of it. Let's see if this page exists under the prefix.
```

Comment:
@juliusv do we need to preserve this functionality. Seems like a vast super-usability feature not even web-centric applications provide in general.